scope_registry.go

```diff
@@ -138,15 +175,15 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 		bucketCache:     parent.bucketCache,
 		done:            make(chan struct{}),
 	}
-	r.subscopes[key] = subscope
-	if _, ok := r.lockedLookup(preSanitizeKey); !ok {
-		r.subscopes[preSanitizeKey] = subscope
+	subscopeBucket.s[key] = subscope
+	if _, ok := r.lockedLookup(subscopeBucket, preSanitizeKey); !ok {
+		subscopeBucket.s[preSanitizeKey] = subscope
 	}
 	return subscope
 }
 
-func (r *scopeRegistry) lockedLookup(key string) (*scope, bool) {
-	ss, ok := r.subscopes[key]
+func (r *scopeRegistry) lockedLookup(subscopeBucket *bucketScope, key string) (*scope, bool) {
```

Comment:
nit: `lookupWithLock` to be consistent with other naming

Reply:
I didn't touch the name, do you still want me to change this?